onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -49,487 +76,553 @@ struct Provider_IDeviceAllocator_Impl : Provider_IDeviceAllocator {
+  const Provider_IndexedSubGraph_MetaDef* Provider_IndexedSubGraph__GetMetaDef(const Provider_IndexedSubGraph* p) override { return reinterpret_cast<const Provider_IndexedSubGraph_MetaDef*>(reinterpret_cast<const IndexedSubGraph*>(p)->GetMetaDef()); }
+
+  // Provider_KernelDef
+  void Provider_KernelDef__operator_delete(Provider_KernelDef* p) override { delete reinterpret_cast<KernelDef*>(p); }
+
+  // Provider_KernelDefBuilder
+  std::unique_ptr<Provider_KernelDefBuilder> Provider_KernelDefBuilder__construct() override { return std::unique_ptr<Provider_KernelDefBuilder>(reinterpret_cast<Provider_KernelDefBuilder*>(new KernelDefBuilder())); }
+  void Provider_KernelDefBuilder__operator_delete(Provider_KernelDefBuilder* p) override { delete reinterpret_cast<KernelDefBuilder*>(p); }
+
+  void Provider_KernelDefBuilder__SetName(Provider_KernelDefBuilder* p, const char* op_name) override { reinterpret_cast<KernelDefBuilder*>(p)->SetName(op_name); }
+  void Provider_KernelDefBuilder__SetDomain(Provider_KernelDefBuilder* p, const char* domain) override { reinterpret_cast<KernelDefBuilder*>(p)->SetDomain(domain); }
+  void Provider_KernelDefBuilder__SinceVersion(Provider_KernelDefBuilder* p, int since_version) override { reinterpret_cast<KernelDefBuilder*>(p)->SinceVersion(since_version); }
+  void Provider_KernelDefBuilder__Provider(Provider_KernelDefBuilder* p, const char* provider_type) override { reinterpret_cast<KernelDefBuilder*>(p)->Provider(provider_type); }
+  void Provider_KernelDefBuilder__TypeConstraint(Provider_KernelDefBuilder* p, const char* arg_name, MLDataType supported_type) override { reinterpret_cast<KernelDefBuilder*>(p)->TypeConstraint(arg_name, supported_type); }
+  void Provider_KernelDefBuilder__TypeConstraint(Provider_KernelDefBuilder* p, const char* arg_name, const std::vector<MLDataType>& supported_types) override { reinterpret_cast<KernelDefBuilder*>(p)->TypeConstraint(arg_name, supported_types); }
+  void Provider_KernelDefBuilder__InputMemoryType(Provider_KernelDefBuilder* p, OrtMemType type, int input_index) override { reinterpret_cast<KernelDefBuilder*>(p)->InputMemoryType(type, input_index); }
+  void Provider_KernelDefBuilder__OutputMemoryType(Provider_KernelDefBuilder* p, OrtMemType type, int input_index) override { reinterpret_cast<KernelDefBuilder*>(p)->OutputMemoryType(type, input_index); }
+  void Provider_KernelDefBuilder__ExecQueueId(Provider_KernelDefBuilder* p, int queue_id) override { reinterpret_cast<KernelDefBuilder*>(p)->ExecQueueId(queue_id); }
+
+  std::unique_ptr<Provider_KernelDef> Provider_KernelDefBuilder__Build(Provider_KernelDefBuilder* p) override {
+    return std::unique_ptr<Provider_KernelDef>(reinterpret_cast<Provider_KernelDef*>(reinterpret_cast<KernelDefBuilder*>(p)->Build().release()));
```

Comment:
>release [](start = 133, length = 7)

I think this function needs to be broken up in more lines.

Reply:
Wouldn't that be bad? It's transferring ownership into a new unique_ptr, if I break it up into multiple lines there'll be the possibility of inserting ways to fail mid ownership transfer (if anyone else adds new lines of code that could fail). As one line it's less likely, plus it's all reinterpret_casts that make it big.